machines/biometrics.ts

```diff
@@ -0,0 +1,227 @@
+
+      initEnrolled: {
+        invoke: {
+          src: LocalAuthentication.isEnrolledAsync,
+          onError: 'failure',
+          onDone: {
+            target: 'checking',
+            actions: assign({ isEnrolled: (context, event) => event.data }),
+          }
+        }
+      },
+
+      // Checks whether we need to proceed if its available otherwise it gets to failure
+      checking: {
+        always: [
+          {
+            target: 'available',
+            cond: 'checkIfAvailable'
+          },
+          {
+            target: 'failure.unavailable',
+            cond: 'checkIfUnavailable'
+          },
+          {
+            target: 'failure.unenrolled',
+            cond: 'checkIfUnenrolled'
+          }
+        ]
+      },
+
+
+      // if available then wait for any event
+      available: {
+        on: {
+          AUTHENTICATE: 'authenticating'
+        }
+      },
+
+      // authenticating biometrics
+      authenticating: {
+        invoke: {
+          src: () => async () => {
+            let res = await LocalAuthentication.authenticateAsync({
+              promptMessage: 'Biometric Authentication',
+
+              // below can only works for IOS not android
+              // disableDeviceFallback: true,
+              // fallbackLabel: 'Invalid fingerprint attempts, Please try again.'
+            })
+            return res.success;
+          },
+          onError: 'failure',
+          onDone: {
+            target: 'authentication',
+            actions: assign({ status: (context, event) => event.data })
+          }
+        },
+      },
+
+      reauthenticating: {
+        always: [
+          {
+            target: 'authenticating',
+            cond: 'checkIfAvailable'
+          },
+          {
+            target: 'failure.unenrolled'
+          }
+        ]
+      },
+
+      // checks authentication status
+      authentication: {
+        always: [
+          {
+            target: 'success',
+            cond: ctx => ctx.status
+          },
+          {
+            target: 'failure.failed',
+            cond: ctx => !ctx.status
+          }
+        ]
+      },
+
+      success: {
+        meta: 'Success authentication with Biometrics',
+      },
+
+      failure: {
+        initial: 'error',
+        states: {
+          unavailable: {
+            meta: 'Device does not support Biometrics'
+          },
+          unenrolled: {
+            meta: 'To use Biometrics, please enroll your fingerprint in your device settings',
+            on: {
+              RETRY_AUTHENTICATE: {
+                actions: assign({retry: (context, event) => true}),
+                target: [
+                  '#biometrics.initEnrolled',
+                  '#biometrics.reauthenticating'
+                ]
+              }
+            }
+          },
+          failed: {
+            after: {
+              // after 1 seconds, transition to available
+              1000: { target: '#biometrics.available' }
```

Comment:
Shorthand for this is `1000: '#biometrics.available` (if you only have `target` you can use a string instead of an object)

Reply:
Cool, ill try that